tests/test_iqtree/test_fit_tree.py

```diff
@@ -26,3 +26,21 @@ def test_fit_tree(three_otu: ArrayAlignment, iq_model: DnaModel, c3_model: str)
     # Should be within an approximation for any seed
     got2 = piqtree2.fit_tree(three_otu, tree_topology, Model(iq_model), rand_seed=None)
     assert got2.params["lnL"] == pytest.approx(expected.lnL)
+
+    # Check motif and rate parameters
+    expected_motif_prob = expected.tree.params["mprobs"]
+    for k, v in expected_motif_prob.items():
+        assert k in got2.params["mprobs"]
+        assert got2.params["mprobs"][k] == pytest.approx(v)
+
+    exclude = {"length", "ENS", "paralinear", "mprobs"}
+    expected_rates = {
+        k: v
+        for k, v in expected.tree.get_edge_vector()[0].params.items()
+        if k not in exclude
+    }
+    for k, v in expected_rates.items():
+        assert k in got2.get_edge_vector()[0].params
+        assert got2.get_edge_vector()[0].params[k] == pytest.approx(
+            v, abs=1e-1,
```

Comment:
i think this value too permissive, can you remove and let me see the failures

Reply:
> i think this value too permissive, can you remove and let me see the failures

Hi @GavinHuttley , I've removed the tolerance setting and brought out the failures. Each fit of K80, GTR, TN93 and HKY85 showed one rate parameter that did not match the expected.